ledger/participant-integration-api/src/main/scala/platform/indexer/ExecuteUpdate.scala

```diff
@@ -243,63 +256,98 @@ trait ExecuteUpdate {
-              "updateCommandId" -> info.commandId,
-              "updateDeduplicateUntil" -> info.deduplicateUntil.toString,
+              Logging.submitter(info.actAs),
+              Logging.applicationId(info.applicationId),
+              Logging.commandId(info.commandId),
+              Logging.deduplicateUntil(info.deduplicateUntil),
             )
           )
           .getOrElse(Map.empty)
       case CommandRejected(_, submitterInfo, reason) =>
         Map(
-          "updateSubmitter" -> loggingContextPartiesValue(submitterInfo.actAs),
-          "updateApplicationId" -> submitterInfo.applicationId,
-          "updateCommandId" -> submitterInfo.commandId,
-          "updateDeduplicateUntil" -> submitterInfo.deduplicateUntil.toString,
-          "updateRejectionReason" -> reason.description,
+          Logging.submitter(submitterInfo.actAs),
+          Logging.applicationId(submitterInfo.applicationId),
+          Logging.commandId(submitterInfo.commandId),
+          Logging.deduplicateUntil(submitterInfo.deduplicateUntil),
+          Logging.rejectionReason(reason.description),
         )
     }
 
-  private def loggingContextPartiesValue(parties: List[Party]): String =
-    parties.mkString("[", ", ", "]")
+  private object Logging {
+    def submissionId(id: SubmissionId): (String, String) =
+      "submissionId" -> id
+    def submissionIdOpt(id: Option[SubmissionId]): (String, String) =
+      "submissionId" -> id.getOrElse("")
+    def participantId(id: ParticipantId): (String, String) =
+      "participantId" -> id
+    def commandId(id: CommandId): (String, String) =
+      "commandId" -> id
+    def party(party: Party): (String, String) =
+      "party" -> party
+    def transactionId(id: TransactionId): (String, String) =
+      "transactionId" -> id
+    def applicationId(id: ApplicationId): (String, String) =
+      "applicationId" -> id
+    def workflowIdOpt(id: Option[WorkflowId]): (String, String) =
+      "workflowId" -> id.getOrElse("")
+    def ledgerTime(time: Timestamp): (String, String) =
+      "ledgerTime" -> time.toInstant.toString
+    def submissionTime(time: Timestamp): (String, String) =
+      "submissionTime" -> time.toInstant.toString
+    def configGeneration(generation: Long): (String, String) =
+      "configGeneration" -> generation.toString
+    def maxDeduplicationTime(time: Duration): (String, String) =
+      "maxDeduplicationTime" -> time.toString
+    def deduplicateUntil(time: Instant): (String, String) =
+      "deduplicateUntil" -> time.toString
+    def rejectionReason(reason: String): (String, String) =
+      "rejectionReason" -> reason
+    def displayName(name: String): (String, String) =
+      "displayName" -> name
+    def sourceDescriptionOpt(description: Option[String]): (String, String) =
+      "sourceDescription" -> description.getOrElse("")
+    def submitter(parties: List[Party]): (String, String) =
+      "submitter" -> parties.mkString("[", ", ", "]")
+
+  }
```

Comment:
I've created this object to ensure consistent logging for different keys. This is very similar to `com.daml.platform.apiserver.services.logging`. I wonder if it would be worth merging both of these helper utils. What do you think @mziolekda ?

Reply:
I am ok with splitting it to a follow-up PR. This one is getting big as it stands.